app.js

```diff
@@ -0,0 +1,122 @@
+/**
+ * Copyright 2017, Google, Inc.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+const express = require('express');
+
+const fs = require('fs');
+const keyFileName = process.env.KEYFILE || 'key.pem';
+const certificateFileName = process.env.CERTFILE || 'cert.pem';
+const privateKey = fs.readFileSync(keyFileName).toString();
+const certificate = fs.readFileSync(certificateFileName).toString();
+const credentials = {key: privateKey, cert: certificate};
+
+const app = express();
+
+const secret = process.env.SECRET; // set by start.sh (npm postinstall)
+const session = require('express-session');
+let MemoryStore = require('memorystore')(session);
+let configuredSession = session({
+  secret: secret,
+  cookie: {secure: true},
+  resave: false,
+  saveUninitialized: true,
+  store: new MemoryStore({
+    checkPeriod: 86400000 // prune expired entries every 24h 
+  })
+});
+app.set('trust proxy', 1); // trust first proxy 
+app.use(configuredSession);
+
+const https = require('https').Server(credentials, app);
+const io = require('socket.io')(https);
+let iosess = require('socket.io-express-session');
+io.use(iosess(configuredSession));
+
+app.use('/src', express.static(__dirname + '/src'));
+app.use('/bower_components', express.static(__dirname + '/src/bower_components'));
```

Comment:
I don't think this and node_modules is necessary anymore if they are included in `/src`

Reply:
It's currently necessary due to the base tag I'm adding to the innermost iframe. We can discuss other options for dealing with that problem in person.

Reply:
however I was able to remove the node_modules one, because (for now) users shouldn't be importing anything from node_modules